diff --git a/setup.py b/setup.py
index 266183a..60dc59f 100644
--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,7 @@ setup(name=__title__,
       long_description=long_desc,
       license=__license__,
       platforms=['linux'],
-      packages=find_packages(exclude=["*.tests"]),
+      packages=find_packages(exclude=["tests"]),
       install_requires=['requests', ],
       requires=['requests', ],
       provides=[__title__, ],
