From dac2da74bf927bb223973b86cb1d95e239aae9bf Mon Sep 17 00:00:00 2001
From: Louis Leseur <louis.leseur@gmail.com>
Date: Sun, 7 Jun 2020 00:02:21 +0200
Subject: [PATCH] Add IOnPrimaryClipChangedListener

Fixes #1481

Signed-off-by: Louis Leseur <louis.leseur@gmail.com>
---
 server/build_without_gradle.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/server/build_without_gradle.sh b/server/build_without_gradle.sh
index f6889701..e3be2faf 100755
--- a/server/build_without_gradle.sh
+++ b/server/build_without_gradle.sh
@@ -42,6 +42,8 @@ echo "Generating java from aidl..."
 cd "$SERVER_DIR/src/main/aidl"
 "$ANDROID_HOME/build-tools/$BUILD_TOOLS/aidl" -o"$CLASSES_DIR" \
     android/view/IRotationWatcher.aidl
+"$ANDROID_HOME/build-tools/$BUILD_TOOLS/aidl" -o"$CLASSES_DIR" \
+    android/content/IOnPrimaryClipChangedListener.aidl
 
 echo "Compiling java sources..."
 cd ../java
@@ -55,6 +57,7 @@ cd "$CLASSES_DIR"
 "$ANDROID_HOME/build-tools/$BUILD_TOOLS/dx" --dex \
     --output "$BUILD_DIR/classes.dex" \
     android/view/*.class \
+    android/content/*.class \
     com/genymobile/scrcpy/*.class \
     com/genymobile/scrcpy/wrappers/*.class
 
