diff --git a/CMakeLists.txt b/CMakeLists.txt
index 303806e..20bad6c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,16 +50,18 @@ if(${COTP_VERSION} LESS 2.0.0)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DCOTP_OLD_LIB")
   pkg_check_modules(BASEENCODE REQUIRED baseencode>=1.0.12)
 endif()
-pkg_check_modules(PNG REQUIRED libpng>=1.2.0)
 pkg_check_modules(JANSSON REQUIRED jansson>=2.10.0)
-pkg_check_modules(ZBAR REQUIRED zbar>=0.20)
 pkg_check_modules(GTK3 REQUIRED gtk+-3.0>=3.20.0)
 pkg_check_modules(GLIB2 REQUIRED glib-2.0>=2.64.0)
 pkg_check_modules(GIO REQUIRED gio-2.0>=2.64.0)
 pkg_check_modules(UUID REQUIRED uuid>=2.34.0)
 pkg_check_modules(PROTOC REQUIRED libprotobuf-c>=1.3.0)
 pkg_check_modules(LIBSECRET REQUIRED libsecret-1>=0.20.0)
-pkg_check_modules(LIBQRENCODE REQUIRED libqrencode>=4.0.2)
+if(BUILD_GUI)
+	pkg_check_modules(PNG REQUIRED libpng>=1.2.0)
+	pkg_check_modules(ZBAR REQUIRED zbar>=0.20)
+	pkg_check_modules(LIBQRENCODE REQUIRED libqrencode>=4.0.2)
+endif()
 
 set(GUI_HEADER_FILES
         src/common/common.h
