--- a/sdl2/Makefile21.unix	2020-01-18 19:18:29.794294743 +0100
+++ b/sdl2/Makefile21.unix	2020-01-18 19:30:16.206921884 +0100
@@ -68,6 +68,8 @@
 CFLAGS   += -DSUPPORT_IA32_HAXM -DUSE_CUSTOM_HOOKINST
 endif
 
+prefix := "/usr/local"
+
 
 all: $(TARGET)
 $(TARGET): $(OBJECTS)
@@ -90,9 +92,10 @@
 
 install:
 	strip $(TARGET)
-	cp $(TARGET) /usr/local/bin/
+	mkdir -p "$(DESTDIR)$(prefix)/bin"
+	cp $(TARGET) "$(DESTDIR)$(prefix)/bin/${TARGET}"
 
 
 uninstall:
-	rm /usr/local/bin/$(TARGET)
+	rm "$(DESTDIR)$(prefix)/bin/$(TARGET)"
 
