From b2a91248c6996341d1161692897562572b8501e5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Sun, 25 Sep 2022 21:14:38 +0200
Subject: [PATCH 1/4] test: Use PEP517 build in tests (via isolated_build=True)

---
 tox.ini | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/tox.ini b/tox.ini
index c8cc4ff..4c1f570 100644
--- a/tox.ini
+++ b/tox.ini
@@ -1,11 +1,10 @@
 [tox]
 envlist = py36, py37, py38
-skipsdist = {env:TOXBUILD:false}
+isolated_build = True
 
 [testenv]
-setenv =
-    PYTHONPATH = {toxinidir}:{toxinidir}/yeelight
-commands = py.test --cov=yeelight --cov-report=term
+commands =
+    pytest --cov=yeelight --cov-report=term
 deps =
     -rrequirements_dev.txt
 
-- 
GitLab


From 59499283ef0a837c41dbe9b84f1b3106ae81b892 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Sun, 25 Sep 2022 21:15:20 +0200
Subject: [PATCH 2/4] Remove obsolete setup.py

---
 setup.py | 43 -------------------------------------------
 1 file changed, 43 deletions(-)
 delete mode 100755 setup.py

diff --git a/setup.py b/setup.py
deleted file mode 100755
index 3d5af94..0000000
--- a/setup.py
+++ /dev/null
@@ -1,42 +0,0 @@
-#!/usr/bin/env python
-# NOTE: This file is only used for tox to be able to install the library to test it.
-# See pyproject.toml for the actual setup file.
-import sys
-
-from setuptools import setup
-
-with open("yeelight/version.py") as f:
-    exec(f.read())
-
-classifiers = [
-    "License :: OSI Approved :: BSD License",
-    "Programming Language :: Python",
-    "Programming Language :: Python :: 3.4",
-    "Programming Language :: Python :: 3.5",
-    "Programming Language :: Python :: 3.6",
-    "Programming Language :: Python :: 3.7",
-    "Programming Language :: Python :: 3.8",
-    "Topic :: Software Development :: Libraries :: Python Modules",
-]
-
-setup(
-    name="yeelight",
-    version=__version__,  # type: ignore
-    author="Stavros Korokithakis",
-    author_email="hi@stavros.io",
-    url="https://gitlab.com/stavros/python-yeelight/",
-    description="A Python library for controlling YeeLight RGB bulbs.",
-    long_description=open("README.md").read(),
-    long_description_content_type="text/markdown",
-    license="BSD",
-    classifiers=classifiers,
-    packages=["yeelight"],
-    python_requires=">=3.4",
-    install_requires=[
-        "future",
-        "ifaddr",
-        'pypiwin32;platform_system=="Windows"',
-    ],
-    test_suite="yeelight.tests",
-    tests_require=[],
-)
GitLab


From 76d6e6fe4d680640bdbe3f8ab98538b4fed07113 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Sun, 25 Sep 2022 21:25:13 +0200
Subject: [PATCH 3/4] Migrate to new-style flit metadata

---
 pyproject.toml | 51 ++++++++++++++++++++++++++++----------------------
 1 file changed, 29 insertions(+), 22 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 1926511..dec374d 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,28 +1,17 @@
-[tool.poetry]
-name = "yeelight"
-version = "0.7.10"
-
 [build-system]
-requires = ["flit_core >=2,<3"]
+requires = ["flit_core >=3.2,<4"]
 build-backend = "flit_core.buildapi"
 
-[tool.flit.metadata]
-module = "yeelight"
-author = "Stavros Korokithakis"
-author-email = "hi@stavros.io"
-home-page = "https://gitlab.com/stavros/python-yeelight"
-description-file = "README.md"
+[project]
+name = "yeelight"
+version = "0.7.10"
+dynamic = ["description"]
+readme = "README.md"
 requires-python = ">=3.4"
-requires-extra = {"dev" = [
-    "flake8",
-    "flake8-docstrings",
-    "flake8-import-order",
-    "flake8-tidy-imports",
-    "pep8-naming",
-    "Sphinx",
-    "sphinx-rtd-theme"
-]}
-requires = ["future", "ifaddr"]
+license = {file = "LICENSE"}
+authors = [
+    {name = "Stavros Korokithakis", email = "hi@stavros.io"},
+]
 classifiers = [
     "License :: OSI Approved :: BSD License",
     "Programming Language :: Python",
@@ -31,5 +20,23 @@ classifiers = [
     "Programming Language :: Python :: 3.6",
     "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
-    "Topic :: Software Development :: Libraries :: Python Modules"
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+dependencies = [
+    "future",
+    "ifaddr",
+]
+
+[project.urls]
+Homepage = "https://gitlab.com/stavros/python-yeelight"
+
+[project.optional-dependencies]
+dev = [
+    "flake8",
+    "flake8-docstrings",
+    "flake8-import-order",
+    "flake8-tidy-imports",
+    "pep8-naming",
+    "Sphinx",
+    "sphinx-rtd-theme",
 ]
-- 
GitLab


From 17886c1ab56b573fdab5d61ef0178dab77f271eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Tue, 27 Sep 2022 08:24:35 +0200
Subject: [PATCH 4/4] Read version from package metadata

---
 pyproject.toml       |  1 +
 setup.cfg            | 10 ----------
 yeelight/__init__.py |  1 -
 yeelight/metadata.py | 13 ++++++++-----
 yeelight/version.py  |  3 ---
 5 files changed, 9 insertions(+), 19 deletions(-)
 delete mode 100644 yeelight/version.py

diff --git a/pyproject.toml b/pyproject.toml
index dec374d..cdb76b3 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,6 +25,7 @@ classifiers = [
 dependencies = [
     "future",
     "ifaddr",
+    "importlib_metadata; python_version<'3.8'",
 ]
 
 [project.urls]
diff --git a/setup.cfg b/setup.cfg
index 78af132..28a818e 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,13 +1,3 @@
-[bumpversion]
-tag = True
-commit = True
-current_version = 0.5.0
-
-[bumpversion:file:yeelight/version.py]
-
-[semantic_release]
-version_variable = yeelight/version.py:__version__
-
 [flake8]
 exclude = docs/*,setup.py,.*
 ignore = F403,E128,E231,E126,E123,E121,E265,E501,N802,N803,N806,C901,D100,D101,D102,D105,W503
diff --git a/yeelight/__init__.py b/yeelight/__init__.py
index 10998bd..1697bd8 100644
--- a/yeelight/__init__.py
+++ b/yeelight/__init__.py
@@ -13,4 +13,3 @@ from yeelight.flow import TemperatureTransition
 from yeelight.main import Bulb
 from yeelight.main import BulbException
 from yeelight.main import discover_bulbs
-from yeelight.version import __version__
diff --git a/yeelight/metadata.py b/yeelight/metadata.py
index d48836a..ba0fd08 100644
--- a/yeelight/metadata.py
+++ b/yeelight/metadata.py
@@ -4,16 +4,19 @@ Project metadata.
 
 Information describing the project.
 """
-from .version import __version__
+try:
+    from importlib.metadata import metadata
+except ImportError:
+    from importlib_metadata import metadata  # type: ignore
+
 
 # The package name, which is also the "UNIX name" for the project.
 package = "yeelight"
 project = "python-yeelight"
 project_no_spaces = project.replace(" ", "")
-version = __version__
-description = (
-    "yeelight is a Python library for controlling YeeLight WiFi RGB" " LED bulbs."
-)
+_package_metadata = metadata(package)
+version = _package_metadata["Version"]
+description = _package_metadata["Summary"]
 authors = ["Stavros Korokithakis"]
 authors_string = ", ".join(authors)
 emails = ["hi@stavros.io"]
diff --git a/yeelight/version.py b/yeelight/version.py
deleted file mode 100644
index 6771014..0000000
--- a/yeelight/version.py
+++ /dev/null
@@ -1,3 +0,0 @@
-# flake8: noqa
-
-__version__ = "0.7.10"
-- 
GitLab

