--- a/sdl2/Makefile.unix	2020-01-18 19:26:02.543842135 +0100
+++ b/sdl2/Makefile.unix	2020-01-18 19:30:22.423582336 +0100
@@ -40,6 +40,8 @@
 CFLAGS   += -DSUPPORT_NET -DSUPPORT_LGY98
 endif
 
+prefix := "/usr/local"
+
 
 all: $(TARGET)
 $(TARGET): $(OBJECTS)
@@ -62,9 +64,10 @@
 
 install:
 	strip $(TARGET)
-	cp $(TARGET) /usr/local/bin/
+	mkdir -p "$(DESTDIR)$(prefix)/bin"
+	cp $(TARGET) "$(DESTDIR)$(prefix)/bin/${TARGET}"
 
 
 uninstall:
-	rm /usr/local/bin/$(TARGET)
+	rm "$(DESTDIR)$(prefix)/bin/${TARGET}"
 
