diff --git a/third_party/crc32c/BUILD.gn b/third_party/crc32c/BUILD.gn
index 5ea81e4..31886e0 100644
--- a/third_party/crc32c/BUILD.gn
+++ b/third_party/crc32c/BUILD.gn
@@ -40,11 +40,7 @@ config("crc32c_config") {
     defines += [ "HAVE_BUILTIN_PREFETCH=0" ]
   }
 
-  if (current_cpu == "arm64") {
-    defines += [ "HAVE_ARM64_CRC32C=1" ]
-  } else {
-    defines += [ "HAVE_ARM64_CRC32C=0" ]
-  }
+  defines += [ "HAVE_ARM64_CRC32C=0" ]
 
   # Android added <sys/auxv.h> in API level 18.
   if (is_linux || is_chromeos || is_android) {
