diff --git a/src/database.c b/src/database.c
index e930f10..35a51ca 100644
--- a/src/database.c
+++ b/src/database.c
@@ -1551,7 +1551,7 @@ int apk_db_open(struct apk_database *db, struct apk_db_options *dbopts)
 	apk_db_setup_repositories(db, dbopts->cache_dir);
 
 	db->cache_max_age = dbopts->cache_max_age ?: 4*60*60; /* 4 hours default */
-	db->root = strdup(dbopts->root ?: "/");
+	db->root = strdup(dbopts->root ?: APK_DEFAULT_ROOT);
 	if (!strcmp(db->root, "/")) db->no_chroot = 1; /* skip chroot if root is default */
 	db->root_fd = openat(AT_FDCWD, db->root, O_RDONLY | O_CLOEXEC);
 	if (db->root_fd < 0 && (dbopts->open_flags & APK_OPENF_CREATE)) {
