diff --git a/cpp/ycm/CMakeLists.txt b/cpp/ycm/CMakeLists.txt
index d7a55a9c..4b4c52b9 100644
--- a/cpp/ycm/CMakeLists.txt
+++ b/cpp/ycm/CMakeLists.txt
@@ -362,19 +362,6 @@ if( LIBCLANG_TARGET )
     file( REMOVE ${FILEPATH} )
   endforeach()
   # When building with MSVC, we need to copy libclang.dll instead of libclang.lib
-  if( MSVC )
-    add_custom_command(
-      TARGET ${PROJECT_NAME}
-      POST_BUILD
-      COMMAND ${CMAKE_COMMAND} -E copy "${PATH_TO_LLVM_ROOT}/bin/libclang.dll" "${LIBCLANG_DIR}"
-    )
-  else()
-    add_custom_command(
-      TARGET ${PROJECT_NAME}
-      POST_BUILD
-      COMMAND ${CMAKE_COMMAND} -E copy "${LIBCLANG_TARGET}" "${LIBCLANG_DIR}"
-    )
-  endif()
 endif()
 
 
@@ -404,9 +391,6 @@ if ( WIN32 OR CYGWIN OR MSYS )
   endforeach()
 endif()
 
-set_target_properties( ${PROJECT_NAME} PROPERTIES
-  LIBRARY_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/../.. )
-
 #############################################################################
 
 if ( USE_DEV_FLAGS AND ( CMAKE_COMPILER_IS_GNUCXX OR COMPILER_IS_CLANG ) )
