diff --git a/ycmd/tests/bindings/cpp_bindings_raises_exception_test.py b/ycmd/tests/bindings/cpp_bindings_raises_exception_test.py
index b18e23f6..ed7f4f51 100644
--- a/ycmd/tests/bindings/cpp_bindings_raises_exception_test.py
+++ b/ycmd/tests/bindings/cpp_bindings_raises_exception_test.py
@@ -21,7 +21,11 @@ from hamcrest import assert_that, calling, raises
 from unittest import TestCase
 ycm_core = ImportCore()
 
-READONLY_MESSAGE = 'can\'t set attribute'
+from sys import version_info
+if version_info >= (3, 11):
+    READONLY_MESSAGE = "property of '.*' object has no setter"
+else:
+    READONLY_MESSAGE = "can't set attribute"
 
 
 class CppBindingsExceptionTest( TestCase ):
