diff --git a/Onboard/pypredict/lm/lm.cpp b/Onboard/pypredict/lm/lm.cpp
index 2e64296..d598280 100644
--- a/Onboard/pypredict/lm/lm.cpp
+++ b/Onboard/pypredict/lm/lm.cpp
@@ -19,7 +19,12 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#ifdef __GLIBC__
 #include <error.h>
+#else
+#include <err.h>
+#define error err
+#endif // __GLIBC__
 #include <algorithm>
 #include <cmath>
 #include <string>
diff --git a/Onboard/pypredict/lm/lm_dynamic.cpp b/Onboard/pypredict/lm/lm_dynamic.cpp
index 7c62824..87e9b0d 100644
--- a/Onboard/pypredict/lm/lm_dynamic.cpp
+++ b/Onboard/pypredict/lm/lm_dynamic.cpp
@@ -17,7 +17,12 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifdef __GLIBC__
 #include <error.h>
+#else
+#include <err.h>
+#define error err
+#endif // __GLIBC__
 
 #include "lm_dynamic.h"
 
